flyway-core/src/main/java/org/flywaydb/core/internal/database/cockroachdb/CockroachDBTable.java

```diff
@@ -70,39 +71,60 @@ public Boolean call() throws SQLException {
     }
 
     protected boolean doExistsOnce() throws SQLException {
-        if (schema.cockroachDB1) {
+        if (schema.cockroachDB1 ) {
             return jdbcTemplate.queryForBoolean("SELECT EXISTS (\n" +
                     "   SELECT 1\n" +
                     "   FROM   information_schema.tables \n" +
                     "   WHERE  table_schema = ?\n" +
                     "   AND    table_name = ?\n" +
                     ")", schema.getName(), name);
+        } else if ( !schema.hasSchemaSupport ) {
+            return jdbcTemplate.queryForBoolean("SELECT EXISTS (\n" +
+                    "   SELECT 1\n" +
+                    "   FROM   information_schema.tables \n" +
+                    "   WHERE  table_catalog = ?\n" +
+                    "   AND    table_schema = 'public'\n" +
+                    "   AND    table_name = ?\n" +
+                    ")", schema.getName(), name);
+        } else {
+            // There is a bug in CockroachDB v20.2.0-beta.* which causes the string equality operator to not work as
+            // expected, therefore we apply a workaround using the like operator.
+            // https://github.com/cockroachdb/cockroach/issues/55437
+            String sql = "SELECT EXISTS (\n" +
+                    "   SELECT 1\n" +
+                    "   FROM   information_schema.tables \n" +
+                    "   WHERE  table_schema = ?\n" +
+                    "   AND    table_name like '%"+name+"%' and length(table_name) = length(?)\n" +
+                    ")";
+            return jdbcTemplate.queryForBoolean(sql, schema.getName(), name);
         }
 
-        return jdbcTemplate.queryForBoolean("SELECT EXISTS (\n" +
-                "   SELECT 1\n" +
-                "   FROM   information_schema.tables \n" +
-                "   WHERE  table_catalog = ?\n" +
-                "   AND    table_schema = 'public'\n" +
-                "   AND    table_name = ?\n" +
-                ")", schema.getName(), name);
     }
 
     @Override
     protected void doLock() throws SQLException {
-        if (lockDepth == 0) {
-            insertRowLock.doLock(jdbcTemplate, database.getInsertStatement(this), database.getBooleanTrue());
+        if ( database.getVersion().isAtLeast("20.1") ) {
+            jdbcTemplate.execute("SELECT * FROM " + this + " FOR UPDATE");
```

Comment:
I am not sure if this will work as desired, since the migration is not run inside of a transaction. (the lock from a SFU update is only held for the duration of a transaction.)

Reply:
OK, I will revert that optimization and always use the `insertRowLock` instead.